pubspec.yaml

```
@@ -16,7 +16,7 @@ dependencies:
   charcode: ^1.2.0
   cli_repl: ^0.2.1
   collection: ^1.16.0
-  http: ">=0.13.3 <1.0.0"
+  http: ">=0.13.3 <2.0.0"
```

```suggestion
  http: "^1.1.0"
```

@Goodwine I think the reason we used `>=0.13.3 <1.0.0` is because `^0.13.3` is equivalent to `>=0.13.3 <0.14.0`, which is too narrow to be practical. Now it would be better to write this as `^1.1.0` if we want to upgrade to 1.x.

---

Right, but it makes me wonder why choose `>0.13 <1.0` instead of changing to `^0.14` like we do with the rest. to me that means there's probably a bug or breaking change in v1, was there?

---

caret syntax has special meaning for `0.x` where `^0.x` is always `>=0.x <0.(x+1)`, because it is assumed 0.x is always unstable.

Here is the description for caret range copied from npm doc:

> Allows changes that do not modify the left-most non-zero digit in the [major, minor, patch] tuple. In other words, this allows patch and minor updates for versions 1.0.0 and above, patch updates for versions 0.X >=0.1.0, and no updates for versions 0.0.X.

---

I see, so it wasn't done to avoid upgrading to v1 for a bug or anything, but rather to avoid this specific case with semver. sgtm